x-pack/apm-server/aggregation/txmetrics/aggregator.go

```diff
@@ -141,16 +160,28 @@ func (a *Aggregator) Run() error {
 		}
 	}()
 	var stop bool
+	var ticks uint64
 	for !stop {
+		ticks++
 		select {
 		case <-a.stopping:
 			stop = true
 		case <-ticker.C:
 		}
-		if err := a.publish(context.Background()); err != nil {
-			a.config.Logger.With(logp.Error(err)).Warnf(
-				"publishing transaction metrics failed: %s", err,
-			)
+		// Publish the metricsets for all configured intervals.
+		for _, interval := range a.intervals {
+			// Publish $interval MetricSets when:
+			//  - ticks * MetricsInterval % $interval == 0.
+			//  - Aggregator is stopped.
+			if !stop && (ticks*uint64(a.config.MetricsInterval))%uint64(interval) != 0 {
+				continue
+			}
```

Comment:
I did have that check initially and removed it since we aren't planning  to allow users to tweak that setting it seemed superfluous, I'm happy adding it back.